nrf_802154/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+#
+# Configurable build parameters
+#
+set(NRF52_SERIES  OFF CACHE BOOL   "Set to TRUE if target is an nRF52 series SoC.")
+set(NRF53_SERIES  OFF CACHE BOOL   "Set to TRUE if target is an nRF53 series SoC.")
+set(RADIO_DRIVER  OFF CACHE BOOL   "Set to TRUE if target requires the radio driver.")
+set(SERIALIZATION OFF CACHE BOOL   "Set to TRUE if target requires serialization module.")
+set(SER_HOST      OFF CACHE BOOL   "Set to TRUE if target is serialization host. Otherwise it is assumed that the target is serialization remote.")
+set(SL_OPENSOURCE OFF CACHE BOOL   "Set to TRUE to link with open source SL library")
```

Comment:
Are there a specific reason those must be present in the CMake cache ?

Cache variables are variables that should be persistent across multiple build invocation, and usually user controllable.
(In Zephyr, Kconfig is used for the user controllable variables).

To me, it seems `NRF52_SERIES` and `NRF53_SERIES` are not to be controlled directly by a user, but a setting derived from other knowledge.

Not sure on the other settings.

Using a cache variable like this also means any existing non-cache variable of same name are hidden.

`RADIO_DRIVER`, `SERIALIZATION`, and `SER_HOST` looks very generic and thus in risk of colliding with identical named symbols if used outside Zephyr context.

Reply:
Taking https://github.com/zephyrproject-rtos/hal_nordic/pull/93#discussion_r601213807 into context:

My expectation is that this file will never be the root of the build. My primary intention of writing those variables here was to document their presence. I think those lines can be completely removed and required variables may be described in a documentation.

Reply:
I have removed the cache variables.